front/app/modules/commercial/insights/admin/containers/Insights/Details/Inputs/index.tsx

```diff
@@ -53,25 +55,53 @@ const Inputs = ({
   onLoadMore,
   loading,
 }: InputsProps) => {
-  const category = query.category;
-  const previewedInputId = query.previewedInputId;
+  // Query parameters are stringified to avoid non-primary value dependencies in onSearch useCallback
```

Comment:
Keeping the keywords filter in the url (as an array) complicated things a bit but I still think it's valuable to have shareable filters (especially for this page) so decided to continue with this approach.